packages/mui-joy/src/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,281 @@
+
+  // Use a hook instead of onError on the img element to support server-side rendering.
+  const loaded = useLoaded({ ...imgProps, src, srcSet });
+  const hasImg = src || srcSet;
+  const hasImgNotFailing = hasImg && loaded !== 'error';
+
+  const ownerState = {
+    ...props,
+    color,
+    component,
+    size,
+    variant,
+    shape,
+  };
+
+  const classes = useUtilityClasses(ownerState);
+
+  if (hasImgNotFailing) {
+    children = (
+      <AvatarImg alt={alt} src={src} srcSet={srcSet} className={classes.img} {...imgProps} />
+    );
+  } else if (childrenProp != null) {
+    children = childrenProp;
+  } else if (hasImg && alt) {
+    children = alt[0];
+  } else {
+    children = <AvatarFallback className={classes.fallback} />;
+  }
+
+  return (
+    <AvatarRoot
+      as={component}
+      ownerState={ownerState}
+      className={clsx(classes.root, className)}
+      ref={ref}
+      {...other}
+    >
+      {children}
+    </AvatarRoot>
+  );
+}) as OverridableComponent<AvatarTypeMap>;
+
+Avatar.propTypes /* remove-proptypes */ = {
+  // ----------------------------- Warning --------------------------------
+  // | These PropTypes are generated from the TypeScript type definitions |
+  // |     To update them edit TypeScript types and run "yarn proptypes"  |
+  // ----------------------------------------------------------------------
+  /**
+   * Used in combination with `src` or `srcSet` to
+   * provide an alt attribute for the rendered `img` element.
+   */
+  alt: PropTypes.string,
+  /**
+   * Used to render icon or text elements inside the Avatar if `src` is not set.
+   * This can be an element, or just a string.
+   */
+  children: PropTypes.node,
+  /**
+   * @ignore
+   */
+  className: PropTypes.string,
+  /**
+   * The color of the component. It supports those theme colors that make sense for this component.
+   * @default 'neutral'
+   */
+  color: PropTypes /* @typescript-to-proptypes-ignore */.oneOfType([
+    PropTypes.oneOf(['danger', 'info', 'neutral', 'primary', 'success', 'warning']),
+    PropTypes.string,
+  ]),
+  /**
+   * The component used for the root node.
+   * Either a string to use a HTML element or a component.
+   */
+  component: PropTypes.elementType,
+  /**
+   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attributes) applied to the `img` element if the component is used to display an image.
+   * It can be used to listen for the loading error event.
+   */
+  imgProps: PropTypes.object,
+  /**
+   * The shape of the component.
+   * @default 'circlular'
+   */
+  shape: PropTypes /* @typescript-to-proptypes-ignore */.oneOfType([
+    PropTypes.oneOf(['circlular', 'rounded', 'square']),
+    PropTypes.string,
+  ]),
+  /**
+   * The size of the component.
+   * It accepts theme values between 'xs' and 'xl'.
+   * @default 'md'
+   */
+  size: PropTypes /* @typescript-to-proptypes-ignore */.oneOfType([
+    PropTypes.oneOf(['lg', 'md', 'sm']),
```

Comment:
it looks nice shortening the `size` props. It would be nice too if we keep the consistency of `size` props like the other component? 
![image](https://user-images.githubusercontent.com/36581242/156790903-cdf1841d-3993-4f46-8b54-71585bd13247.png)

Reply:
@pahrizal Glad, you like it! this component is for the new design system (Joy UI) we are working on but the image you posted is Material UI, so right now it is a different package. We will try to make both of them consistent to yield the best DX in the future.